Update arrow function documentation to clarify naming and assignment (#34501)
mdn/content: files/en-us/web/javascript/reference/functions/arrow_functions/index.md

};
```

> Arrow functions are always unnamed. If the arrow function needs to call itself, use a named function expression instead. You can also assign the arrow function to a variable, allowing you to refer to it through that variable.

Reviewer: ```suggestion
Arrow functions are not inherently associated with a name. If the arrow function needs to call itself, use a named function expression instead. You can also assign the arrow function to a variable, allowing you to refer to it through that variable.
```
